test_suite/tests/test_annotations.rs

```diff
@@ -2827,55 +2909,190 @@ mod flatten {
+                            variant: "Unit",
+                        },
+                        // content missing
+                        // outer
+                        Token::Str("outer"),
+                        Token::U32(42),
+                        Token::MapEnd,
+                    ],
+                );
+                // Field order: outer, [tag, content]
+                assert_de_tokens(
+                    &value,
+                    &[
+                        Token::Map { len: None },
+                        // outer
+                        Token::Str("outer"),
+                        Token::U32(42),
+                        // tag
+                        Token::Str("tag"),
+                        Token::UnitVariant {
+                            name: "Enum",
+                            variant: "Unit",
                         },
+                        // content
                         Token::Str("content"),
-                        Token::Struct {
-                            len: 2,
-                            name: "Struct",
+                        Token::Unit,
+                        Token::MapEnd,
+                    ],
+                );
+                // Field order: outer, [content, tag]
+                assert_de_tokens(
+                    &value,
+                    &[
+                        Token::Map { len: None },
+                        // outer
+                        Token::Str("outer"),
+                        Token::U32(42),
+                        // content
+                        Token::Str("content"),
+                        Token::Unit,
+                        // tag
+                        Token::Str("tag"),
+                        Token::UnitVariant {
+                            name: "Enum",
+                            variant: "Unit",
                         },
-                        Token::Str("index"),
-                        Token::U32(0),
-                        Token::Str("value"),
+                        Token::MapEnd,
+                    ],
+                );
+                // Field order: [tag, content], outer
+                assert_de_tokens(
+                    &value,
+                    &[
+                        Token::Map { len: None },
+                        // tag
+                        Token::Str("tag"),
+                        Token::UnitVariant {
+                            name: "Enum",
+                            variant: "Unit",
+                        },
+                        // content
+                        Token::Str("content"),
+                        Token::Unit,
+                        // outer
+                        Token::Str("outer"),
                         Token::U32(42),
-                        Token::StructEnd,
+                        Token::MapEnd,
+                    ],
+                );
+                // Field order: [content, tag], outer - did not work
```

Comment:
I am unsure how to interpret this "did not work" comment (and another one below). Is this to indicate something that was unsupported by the original implementation of `flatten` that has since become supported in a more recent release of serde? If so, I am not sure that this comment is warranted and providing clarity.

Reply:
I don't remember this now - it's too many time since creating this PR. Probably that did not work at time of writing and then was fixed by other PR. Currently that part of comment can be safely removed